src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOSupport.scala

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.dsde.firecloud.dataaccess
+
+import java.net.InetAddress
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.broadinstitute.dsde.firecloud.FireCloudException
+import org.elasticsearch.action.{ActionRequest, ActionRequestBuilder, ActionResponse}
+import org.elasticsearch.client.transport.TransportClient
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.transport.InetSocketTransportAddress
+import spray.http.Uri.Authority
+
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Created by davidan on 10/5/16.
+  */
+trait ElasticSearchDAOSupport extends LazyLogging {
+
+  def buildClient(servers:Seq[Authority]): TransportClient = {
+    // cluster name is constant across environments; no need to add it to config
+    val settings = Settings.settingsBuilder
+      .put("cluster.name", "firecloud-elasticsearch")
+      .build
+    val addresses = servers map { server =>
+      new InetSocketTransportAddress(InetAddress.getByName(server.host.address), server.port)
+    }
+    TransportClient.builder().settings(settings).build.addTransportAddresses(addresses: _*)
+  }
+
+  def executeESRequest[T <: ActionRequest[T], U <: ActionResponse, V <: ActionRequestBuilder[T, U, V]](req: V): U = {
```

Comment:
The ES client has its own async/future handling, based on java.util.concurrent.Future (and heavily subclassed). I did intend to look at it, but was daunted by the possible conflicts with Scala futures ... and then never created a follow-on story or mentioned it in the PR description. I'll make a story.